pkg/didcomm/protocol/didexchange/states.go

```diff
@@ -174,6 +174,8 @@ func (s *requested) ExecuteInbound(msg *stateMachineMsg, thid string, ctx *conte
 		if err != nil {
 			return nil, nil, nil, fmt.Errorf("JSON unmarshalling of invitation: %w", err)
 		}
+		// sets invitation
+		msg.invitation = invitation
```

Comment:
modifying input parameters is generally bad practice.

Reply:
I did this to improve performance. Otherwise, we need to unmarshal the message every time.